pkg/generators/rules/names_match.go

```diff
@@ -109,14 +100,30 @@ func (n *NamesMatch) Validate(t *types.Type) ([]string, error) {
 				continue
 			}
 			jsonName := strings.Split(jsonTag, ",")[0]
-			if !namesMatch(goName, jsonName) {
+			if !nameIsOkay(m, jsonName) {
 				fields = append(fields, goName)
 			}
 		}
 	}
 	return fields, nil
 }
 
+func nameIsOkay(member types.Member, jsonName string) bool {
+	if jsonName == "" {
+		return member.Type.Kind == types.Struct ||
+			member.Type.Kind == types.Pointer && member.Type.Underlying.Kind == types.Struct
```

Comment:
Ooops. This should be member.Type.Elem.

/cancel lgtm

Reply:
https://github.com/kubernetes/kubernetes/blob/5ac315faf428567ec023aa7d8c2b89b1d9c9211a/vendor/k8s.io/gengo/v2/types/types.go#L327-L335

Reply:
Noticed just in time when looking at the gengo API changes, but missed it earlier when writing the code... sorry!